pcf/particle/azure/storage/blob.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pcf.core.azure_resource import AzureResource
+from pcf.core import State
+from azure.storage.blob import PublicAccess
+
+
+class Blob(AzureResource):
+    """
+    This is the implementation of Azure's blob container service.
+    https://azure-storage.readthedocs.io/ref/azure.storage.blob.html
+    """
+    flavor = "blob"
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    UNIQUE_KEYS = ["azure_resource.name"]
+
+    def __init__(self, particle_definition, blob_service=None):
```

Comment:
pass in parameter client into azure_resource so in this case would be `storage_client` .